libs/common-const/src/networks.ts

```diff
@@ -1,32 +1,41 @@
 import { SupportedChainId } from '@cowprotocol/cow-sdk'
 import { JsonRpcProvider } from '@ethersproject/providers'
 
-function initRpcUrls(): Record<SupportedChainId, string> {
-  const REACT_APP_INFURA_KEY = process.env.REACT_APP_INFURA_KEY
-  const REACT_APP_NETWORK_URL_1 = process.env.REACT_APP_NETWORK_URL_1
-  const REACT_APP_NETWORK_URL_5 = process.env.REACT_APP_NETWORK_URL_5
-  const REACT_APP_NETWORK_URL_100 = process.env.REACT_APP_NETWORK_URL_100
-
-  if (!REACT_APP_INFURA_KEY && !(REACT_APP_NETWORK_URL_1 && REACT_APP_NETWORK_URL_5 && REACT_APP_NETWORK_URL_100)) {
-    throw new Error(
-      `Either REACT_APP_INFURA_KEY or REACT_APP_NETWORK_URL_{1,5,100} environment variables must be defined`
-    )
-  }
-
-  return {
-    [SupportedChainId.MAINNET]: REACT_APP_NETWORK_URL_1 || `https://mainnet.infura.io/v3/${INFURA_KEY}`,
-    [SupportedChainId.GNOSIS_CHAIN]: REACT_APP_NETWORK_URL_100 || `https://rpc.gnosis.gateway.fm`,
-    [SupportedChainId.GOERLI]: REACT_APP_NETWORK_URL_5 || `https://goerli.infura.io/v3/${INFURA_KEY}`,
-  }
-}
-
 const INFURA_KEY = process.env.REACT_APP_INFURA_KEY
-if (typeof INFURA_KEY === 'undefined') {
-  throw new Error(`REACT_APP_INFURA_KEY must be a defined environment variable`)
+
+const DEFAULT_RPC_URL: Record<SupportedChainId, { url: string; usesInfura: boolean }> = {
+  [SupportedChainId.MAINNET]: { url: `https://mainnet.infura.io/v3/${INFURA_KEY}`, usesInfura: true },
+  [SupportedChainId.GNOSIS_CHAIN]: { url: `https://rpc.gnosis.gateway.fm`, usesInfura: false },
+  [SupportedChainId.GOERLI]: { url: `https://goerli.infura.io/v3/${INFURA_KEY}`, usesInfura: true },
 }
 
 /**
  * These are the network URLs used by the interface when there is not another available source of chain data
  */
-export const RPC_URLS = initRpcUrls()
+export const RPC_URLS = {
+  [SupportedChainId.MAINNET]: getRpcUrl(SupportedChainId.MAINNET),
+  [SupportedChainId.GNOSIS_CHAIN]: getRpcUrl(SupportedChainId.GNOSIS_CHAIN),
+  [SupportedChainId.GOERLI]: getRpcUrl(SupportedChainId.GOERLI),
+}
+
 export const MAINNET_PROVIDER = new JsonRpcProvider(RPC_URLS[SupportedChainId.MAINNET])
+
+function getRpcUrl(chainId: SupportedChainId): string {
+  const envKey = `REACT_APP_NETWORK_URL_${chainId}`
+  const rpcUrl = process.env[envKey]
+
+  console.log('RPC', { envKey, rpcUrl, INFURA_KEY })
+
+  if (rpcUrl) {
+    return rpcUrl
+  }
+
+  const defaultRpc = DEFAULT_RPC_URL[chainId]
+  if (defaultRpc.usesInfura && !INFURA_KEY) {
+    throw new Error(`Either ${envKey} or INFURA_KEY environment variable are required`)
```

Comment:
Prefix here too, if needed: `REACT_APP_INFURA_KEY`

Reply:
Here it is needed indeed. Fixing!